lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlockBlobFileSystem.cs

```diff
@@ -0,0 +1,205 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.IO.FileSystem.AzureBlob
+{
+    /// <summary>
+    /// An IFileSystem implementation for Azure Blob storage.
+    /// This particular File System implementation only supports 
+    /// block blob operations
+    /// </summary>
+    internal sealed class AzureBlockBlobFileSystem : IFileSystem
+    {
+        private readonly ICloudBlobClient _client;
+
+        [Inject]
+        private AzureBlockBlobFileSystem(ICloudBlobClient client)
+        {
+            _client = client;
+        }
+
+        /// <summary>
+        /// Not supported for Azure Blobs, will throw <see cref="NotSupportedException"/>.
+        /// </summary>
+        public Stream Open(Uri fileUri)
+        {
+            throw new NotSupportedException("Open is not supported for AzureBlockBlobFileSystem.");
```

Comment:
Yes, but we don't need this at the moment, and it is much more complicated and suitable for a separate JIRA because it involves managing blocks locally.